Boolean.CSharp.Test/CoreAndExtensionTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Boolean.CSharp.Main;
+using NUnit.Framework;
+
+namespace Boolean.CSharp.Test
+{
+    public class CoreAndExtensionTest
+    {
+        [Test]
+        public void TestAccount()
+        {
+            var branch = new Branch(1 , "Branch Name" , "Branch Address" , "Branch SortCode");
+            var account = new CurrentAccount(1 , branch);
+
+            Assert.AreEqual(1 , account.GetId());
+            Assert.AreEqual(branch , account.GetBranch());
+            Assert.AreEqual(0 , account.ViewBalance());
+            Assert.AreEqual(0 , account.Transactions.Count);
+
+            Assert.IsTrue(account.Deposit(100));
+            Assert.AreEqual(100 , account.ViewBalance());
+
+            Assert.IsFalse(account.Deposit(-50));
+            Assert.AreEqual(100 , account.ViewBalance());
+
+            Assert.IsTrue(account.Withdraw(50));
+            Assert.AreEqual(50 , account.ViewBalance());
+
+            Assert.IsFalse(account.Withdraw(60));
+            Assert.AreEqual(50 , account.ViewBalance());
+
+            Assert.IsFalse(account.Withdraw(-10));
+            Assert.AreEqual(50 , account.ViewBalance());
+        }
+
+        [Test]
+        public void TestCurrentAccount()
+        {
+            var branch = new Branch(1 , "Branch Name" , "Branch Address" , "Branch SortCode");
+            var account = new CurrentAccount(1 , branch);
+
+            Assert.AreEqual(0 , account.OverdraftLimit);
+
+            branch.AddAccount(account);
+            branch.ApproveOverdraftRequest(1 , 500);
+            Assert.AreEqual(500 , account.OverdraftLimit);
```

Comment:
you are not testing whether, when account balance is 0, you can withdeaw up to 500